src/main/java/com/microsoft/sqlserver/jdbc/SQLServerConnection.java

```diff
@@ -3621,16 +3644,24 @@ final boolean doExecute() throws SQLServerException {
 
     private void logon(LogonCommand command) throws SQLServerException {
         SSPIAuthentication authentication = null;
-        if (integratedSecurity && AuthenticationScheme.nativeAuthentication == intAuthScheme)
-            authentication = new AuthenticationJNI(this, currentConnectPlaceHolder.getServerName(),
-                    currentConnectPlaceHolder.getPortNumber());
-        if (integratedSecurity && AuthenticationScheme.javaKerberos == intAuthScheme) {
-            if (null != ImpersonatedUserCred) {
-                authentication = new KerbAuthentication(this, currentConnectPlaceHolder.getServerName(),
-                        currentConnectPlaceHolder.getPortNumber(), ImpersonatedUserCred, isUserCreatedCredential);
-            } else
-                authentication = new KerbAuthentication(this, currentConnectPlaceHolder.getServerName(),
+
+        if (integratedSecurity) {
+            if (AuthenticationScheme.nativeAuthentication == intAuthScheme) {
+                authentication = new AuthenticationJNI(this, currentConnectPlaceHolder.getServerName(),
                         currentConnectPlaceHolder.getPortNumber());
+            } else if (AuthenticationScheme.javaKerberos == intAuthScheme) {
+                if (null != ImpersonatedUserCred) {
+                    authentication = new KerbAuthentication(this, currentConnectPlaceHolder.getServerName(),
+                            currentConnectPlaceHolder.getPortNumber(), ImpersonatedUserCred, isUserCreatedCredential);
+                } else
+                    authentication = new KerbAuthentication(this, currentConnectPlaceHolder.getServerName(),
+                            currentConnectPlaceHolder.getPortNumber());
+            } else if (ntlmAuthentication) {
+                authentication = new NTLMAuthentication(this, domainName,
+                        activeConnectionProperties.getProperty(SQLServerDriverStringProperty.USER.toString()),
+                        activeConnectionProperties.getProperty(SQLServerDriverStringProperty.PASSWORD.toString()),
+                        hostName);
```

Comment:
Please add `activeConnectionProperties.remove(SQLServerDriverStringProperty.PASSWORD.toString());` to also remove the password from connection properties.